sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnVectorUtils.scala

```diff
@@ -38,4 +40,114 @@ object GpuColumnVectorUtils {
+   *                 should not infer from scalars
+   * @return column vector for the specified scalars
+   */
+  def createFromScalarList(scalars: Seq[GpuScalar], dataType: DataType): CudfCV = {
+    scalars.head.dataType match {
+      case BooleanType =>
+        val booleans = scalars.map(s => s.getValue.asInstanceOf[java.lang.Boolean])
+        CudfCV.fromBoxedBooleans(booleans: _*)
+      case ByteType =>
+        val bytes = scalars.map(s => s.getValue.asInstanceOf[java.lang.Byte])
+        CudfCV.fromBoxedBytes(bytes: _*)
+      case ShortType =>
+        val shorts = scalars.map(s => s.getValue.asInstanceOf[java.lang.Short])
+        CudfCV.fromBoxedShorts(shorts: _*)
+      case IntegerType =>
+        val ints = scalars.map(s => s.getValue.asInstanceOf[java.lang.Integer])
+        CudfCV.fromBoxedInts(ints: _*)
+      case LongType =>
+        val longs = scalars.map(s => s.getValue.asInstanceOf[java.lang.Long])
+        CudfCV.fromBoxedLongs(longs: _*)
+      case FloatType =>
+        val floats = scalars.map(s => s.getValue.asInstanceOf[java.lang.Float])
+        CudfCV.fromBoxedFloats(floats: _*)
+      case DoubleType =>
+        val doubles = scalars.map(s => s.getValue.asInstanceOf[java.lang.Double])
+        CudfCV.fromBoxedDoubles(doubles: _*)
+      case StringType =>
+        val utf8Bytes = scalars.map(s => {
+          val v = s.getValue
+          if (v == null) {
+            null
+          } else {
+            v.asInstanceOf[UTF8String].getBytes
+          }
+        })
+        CudfCV.fromUTF8Strings(utf8Bytes: _*)
+      case DecimalType() =>
+        val decimals = scalars.map(s => {
+          val v = s.getValue
+          if (v == null) {
+            null
+          } else {
+           v.asInstanceOf[Decimal].toJavaBigDecimal
+          }
+        })
+        fromDecimals(dataType.asInstanceOf[DecimalType], decimals: _*)
+      case _ =>
+        throw new UnsupportedOperationException(s"Creating column vector from a GpuScalar list" +
+            s" is not supported for type ${scalars.head.dataType}.")
+    }
+  }
+
+  /**
+   * Create decimal column vector according to DecimalType.
+   * Note: it will create 3 types of column vector according to DecimalType precision
+   *  - Decimal 32 bits
+   *  - Decimal 64 bits
+   *  - Decimal 128 bits
+   * E.g.: If the max of values are decimal 32 bits, but DecimalType is 128 bits,
+   * then return a Decimal 128 bits column vector
+   */
+  def fromDecimals(dt: DecimalType, values: java.math.BigDecimal*): CudfCV = {
+    val hcv = HostColumnVector.build(
+      fromJavaBigDecimal(dt),
+      values.length,
+      new Consumer[HostColumnVector.Builder]() {
+        override def accept(b: HostColumnVector.Builder): Unit = {
+          b.appendBoxed(values: _*)
+        }
+      }
+    )
+    withResource(hcv) { _ =>
+      hcv.copyToDevice()
+    }
+  }
+
+  private def fromJavaBigDecimal(dt: DecimalType): DType = {
```

Comment:
we already have this util here https://github.com/NVIDIA/spark-rapids/blob/branch-24.08/sql-plugin/src/main/scala/com/nvidia/spark/rapids/DecimalUtil.scala#L26